src/components/RichContenteditable/RichContenteditable.vue

```diff
@@ -0,0 +1,328 @@
+  -
+  - @author John Molakvoæ <skjnldsv@protonmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<docs>
+
+### General description
+
+This component displays contenteditable div with automated @ autocompletion [at].
+
+### Examples
+
+```vue
+<template>
+	<div>
+		<RichContenteditable
+			v-model="message"
+			:auto-complete="autoComplete"
+			:user-data="userData"
+			placeholder="Try mentioning the user Test01"
+			style="width: 350px;height: 100px;" />
+		{{ message }}
+	</div>
+</template>
+<script>
+export default {
+	data() {
+		return {
+			message: '',
+			// You need to provide this for the inline
+			// mention to understand what to display or not.
+			userData: {
+				Test01: {
+					icon: 'icon-user',
+					id: 'Test01',
+					label: 'Test01',
+					source: 'users',
+					primary: true
+				},
+				Test02: {
+					icon: 'icon-user',
+					id: 'Test02',
+					label: 'Test02',
+					source: 'users'
+				}
+			}
+		}
+	},
+	methods: {
+		/**
+		* Do your own query to the autocompletion api.
+		* The returned data bellow is a fake data example.
+		* The callback expects the same format returned by the core/autocomplete/get ocs api endpoint!
+		* @param {string} search the query
+		* @param {Function} callback the callback to process the results with
+		*/
+		autoComplete(search, callback) {
+			return callback(Object.values(this.userData))
+		}
+	}
+}
+</script>
+```
+
+</docs>
+
+<template>
+	<div ref="contenteditable"
+		:class="{'rich-contenteditable__input--empty': isEmptyValue}"
+		:contenteditable="contenteditable"
+		:placeholder="placeholder"
+		class="rich-contenteditable__input"
+		@input="onInput"
+		@keydown.delete="onDelete"
+		@paste.stop.prevent="onPaste" />
+</template>
+
+<script>
+import Tribute from 'tributejs/dist/tribute.esm'
+import debounce from 'debounce'
+
+import { t } from '../../l10n.js'
+import AutoCompleteResult from './AutoCompleteResult'
+import richEditor from '../../mixins/richEditor/index'
+
+export default {
+	name: 'RichContenteditable',
+
+	mixins: [richEditor],
+
+	props: {
+		value: {
+			type: String,
+			required: true,
+		},
+		placeholder: {
+			type: String,
+			default: t('Write message, @ to mention someone …'),
+		},
+		autoComplete: {
```

Comment:
should we add a `disabled` property ?

I thought of this when dealing with read-only conversations but went with hiding the field completely. I could imagine cases where it's desirable to display a field, even with contents, but disabled, for example during a XHR "send" operation

Reply:
No, this doesn't work. use `contenteditable="false"`
I prefer to stay as close as the html defaults :)